### PR TITLE
[Feature] Add Case Insensitive Search parameter to annotation

### PR DESCRIPTION
**Pull request by luc-boussant**

### CODE REVIEW

File: src/main/kotlin/com/sipios/springsearch/CriteriaParser.kt [hunk 1]
```diff
@@ -9,9 +9,9 @@ import org.springframework.data.jpa.domain.Specification
 /**
  * Class used to parse a search query string and create a specification
  */
-class CriteriaParser<T> {
+class CriteriaParser<T>(caseSensitiveFlag: Boolean) {
```
Comment (MollardMichael) [inline src/main/kotlin/com/sipios/springsearch/CriteriaParser.kt:12]: I would pass around a reference to the initial annotation (or a context derived from it at least)

File: src/main/kotlin/com/sipios/springsearch/anotation/SearchSpec.kt [hunk 2]
```diff
@@ -11,5 +11,10 @@ annotation class SearchSpec(
     /**
      * The name of the query param that will be transformed into a specification
      */
-    val searchParam: String = "search"
+    val searchParam: String = "search",
+
+    /**
+     * A flag to represent if the search needs to be case sensitive or not
```
Comment (MollardMichael) [inline src/main/kotlin/com/sipios/springsearch/anotation/SearchSpec.kt:17]: represent -> indicate ?

**Review by MollardMichael — CHANGES_REQUESTED** [review]

File: src/main/kotlin/com/sipios/springsearch/anotation/SearchSpec.kt (hunk 2 above)
Comment (luc-boussant) [reply src/main/kotlin/com/sipios/springsearch/anotation/SearchSpec.kt:17]: Done

File: src/main/kotlin/com/sipios/springsearch/CriteriaParser.kt (hunk 1 above)
Comment (luc-boussant) [reply src/main/kotlin/com/sipios/springsearch/CriteriaParser.kt]: I passed around the reference to the initial annotation

File: src/test/kotlin/com/sipios/springsearch/UsersConfig.kt [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.sipios.springsearch
+
+import javax.annotation.PostConstruct
+import org.springframework.context.annotation.Configuration
+
+@Configuration
+class UsersConfig {
```
Comment (MollardMichael) [inline src/test/kotlin/com/sipios/springsearch/UsersConfig.kt:7]: Pourquoi c'est là ?

**Review by MollardMichael — APPROVED** [review]

**Pull request merged**